From baaa13101dbac4bbda91b75186113fe297d2f93d Mon Sep 17 00:00:00 2001
From: Ken Tseng <ken.tseng@realtek.com>
Date: Thu, 8 May 2025 14:51:01 +0800
Subject: [PATCH] recipes-kernel: dts: get optee properties from u-boot

- remove /firmware/optee
- remove /reserved-memory/tee@10000000
-- will be updated as /reserved-memory/optee@10000000
- kent: rescue: keep bl31-mem
---
 arch/arm64/boot/dts/realtek/kent.dtsi                    | 9 ---------
 arch/arm64/boot/dts/realtek/rtd1501-rescue.dtsi          | 5 -----
 arch/arm64/boot/dts/realtek/rtd1501.dtsi                 | 4 ----
 arch/arm64/boot/dts/realtek/rtd1501s-phantom-8gb.dts     | 4 ----
 arch/arm64/boot/dts/realtek/rtd16xxb.dtsi                | 9 ---------
 arch/arm64/boot/dts/realtek/rtd1920-rescue.dtsi          | 5 -----
 arch/arm64/boot/dts/realtek/rtd1920.dtsi                 | 4 ----
 .../boot/dts/realtek/rtd1920s-smallville-4gb-bringup.dts | 4 ----
 8 files changed, 44 deletions(-)

--- a/arch/arm64/boot/dts/realtek/kent.dtsi
+++ b/arch/arm64/boot/dts/realtek/kent.dtsi
@@ -189,10 +189,6 @@
 	};
 
 	firmware: firmware {
-		optee {
-			compatible = "linaro,optee-tz";
-			method = "smc";
-		};
 	};
 
 	gic_extension: gic-extension {
@@ -350,11 +346,6 @@
 			no-map;
 		};
 
-		tee: tee@10000000 {
-			reg = <0x0 0x10000000 0x0 0x04000000>; /* TEE: 64MB */
-			no-map;
-		};
-
 		bootlogo_reserved: buffer@2f700000 {
 			reg = <0x0 0x2f700000 0x0 0x00900000>;
 		};
--- a/arch/arm64/boot/dts/realtek/rtd1501-rescue.dtsi
+++ b/arch/arm64/boot/dts/realtek/rtd1501-rescue.dtsi
@@ -112,10 +112,6 @@
 		no-map;
 	};
 
-	tee@10000000 {
-		status = "okay";
-	};
-
 	bootlogo_reserved: buffer@2f700000 {
 		status = "okay";
 	};
@@ -168,5 +164,4 @@
 
 	/delete-node/linux,cma_1;
 	/delete-node/linux,cma_2;
-	/delete-node/bl31-mem@10001000;
 };
--- a/arch/arm64/boot/dts/realtek/rtd1501.dtsi
+++ b/arch/arm64/boot/dts/realtek/rtd1501.dtsi
@@ -215,10 +215,6 @@
 		no-map;
 	};
 
-	tee: tee@10000000 {
-		status = "disabled";
-	};
-
 	bl31_mem: bl31-mem@10001000 {
 		reg = <0x0 0x10001000 0x0 0x40000>;
 		no-map;
--- a/arch/arm64/boot/dts/realtek/rtd16xxb.dtsi
+++ b/arch/arm64/boot/dts/realtek/rtd16xxb.dtsi
@@ -67,11 +67,6 @@
 			no-map;
 		};
 
-		tee: tee@10000000 {
-			reg = <0x10000000 0x04000000>; /* TEE: 64MB */
-			no-map;
-		};
-
 		vo_reserved: vo_reserved@4800000 {
 			reg = <0x04800000 0x7d000>;
 			no-map;
@@ -350,10 +345,6 @@
 	};
 
 	firmware {
-		optee {
-			compatible = "linaro,optee-tz";
-			method = "smc";
-		};
 	};
 
 	avcert_rproc: avcert-rproc {
--- a/arch/arm64/boot/dts/realtek/rtd1920-rescue.dtsi
+++ b/arch/arm64/boot/dts/realtek/rtd1920-rescue.dtsi
@@ -118,10 +118,6 @@
 		no-map;
 	};
 
-	tee@10000000 {
-		status = "okay";
-	};
-
 	bootlogo_reserved: buffer@2f700000 {
 		status = "okay";
 	};
@@ -174,5 +170,4 @@
 
 	/delete-node/linux,cma_1;
 	/delete-node/linux,cma_2;
-	/delete-node/bl31-mem@10001000;
 };
--- a/arch/arm64/boot/dts/realtek/rtd1920.dtsi
+++ b/arch/arm64/boot/dts/realtek/rtd1920.dtsi
@@ -221,10 +221,6 @@
 		no-map;
 	};
 
-	tee: tee@10000000 {
-		status = "disabled";
-	};
-
 	bl31_mem: bl31-mem@10001000 {
 		reg = <0x0 0x10001000 0x0 0x40000>;
 		no-map;
--- a/arch/arm64/boot/dts/realtek/rtd1920s-smallville-4gb-bringup.dts
+++ b/arch/arm64/boot/dts/realtek/rtd1920s-smallville-4gb-bringup.dts
@@ -102,10 +102,6 @@
 		no-map;
 	};
 
-	tee@10000000 {
-		status = "okay";
-	};
-
 	bootlogo_reserved: buffer@2f700000 {
 		status = "okay";
 	};
